diff --git a/crates/common/src/types/backend_role.rs b/crates/common/src/types/backend_role.rs
new file mode 100644
index 000000000..fdc1f9de5
--- /dev/null
+++ b/crates/common/src/types/backend_role.rs
@@ -0,0 +1,19 @@
+#[derive(Copy, Clone, Debug, PartialEq, Eq, Default, clap::ValueEnum)]
+pub enum BackendRole {
+    #[default]
+    Leader,
+    Follower,
+}
+
+impl BackendRole {
+    pub fn is_leader(&self) -> bool {
+        matches!(self, BackendRole::Leader)
+    }
+
+    pub fn as_str(&self) -> &'static str {
+        match self {
+            BackendRole::Leader => "leader",
+            BackendRole::Follower => "follower",
+        }
+    }
+}
diff --git a/crates/common/src/types/mod.rs b/crates/common/src/types/mod.rs
index 050452ca1..91d89eaa1 100644
--- a/crates/common/src/types/mod.rs
+++ b/crates/common/src/types/mod.rs
@@ -17,6 +17,7 @@ pub use value::{
 mod actions;
 mod admin_key;
 mod backend_info;
+mod backend_role;
 mod backend_state;
 mod deployments;
 mod environment_variables;
@@ -51,6 +52,7 @@ pub use backend_info::{
     BackendInfo,
     DEFAULT_PROVISION_CONCURRENCY,
 };
+pub use backend_role::BackendRole;
 pub use backend_state::{
     BackendState,
     OldBackendState,
